### ISSUE TITLE
Playlist export in m3u/m3u8 fails

### ISSUE THREAD

**Issue by jmfergeau** [mixxxdj/mixxx#11780]
### Bug Description

When a playlist is exported, the m3u and m3u8 generated are just a simple line saying `#EXTM3U`. It doesn't export the files names as intended and then makes the generated playlist completely useless.

### Version

2.3.5

### OS

Arch Linux with LTS kernel

**Comment by daschuer** [comment]
I cannot confirm this with 2.3.5 or HEAD of the 2.3 branch. 
Does the mixxx.log contain something suspicious?

**Comment by daschuer** [comment]
Can you remember the condition when it worked?

**Comment by daschuer** [comment]
Interesting:
```
Warning [Main]: Could not find filter "*.m3u|M3U Playlist "
```

```
Debug [Main]: item not found for QUrl("file:///home/max/Bureau/default.m3u")
```

Why has the file the txt extension? Was it found on the harddisk like that?

**Comment by jmfergeau** [comment]
> Why has the file the txt extension? Was it found on the harddisk like that?

No, it was in m3u. I changed it to txt for upload because github is too stupid to understand m3u files are text files...

**Comment by ronso0** [comment]
@jmfergeau Out of curiosity: how is crate export working? And playlist **import**?

**Comment by jmfergeau** [comment]
> @jmfergeau Out of curiosity: how is crate export working? And playlist **import**?

To export, i simply click the mixxx playlist and click "export playlist"

I haven't tested import.

**Comment by ronso0** [comment]
To be more clear:
is **crate** export working? You can export crates to popular playlist formats, as well.

Please also test playlist **import**.
I hooe that helps narrow this down to some specific part of QFileDialog, or Mixxx.

What Qt version are you using? It is printed in the log stream, during startup.

**Comment by daschuer** [comment]
Mixxx 2.3.5 with Qt: 5.15.10 on Arch Linux. 

This is the relevant part form the log: 
```
Debug [Main]: Already focused on playlist  -1
Warning [Main]: BaseSqlTableModel::setSort invalid column: -1
Debug [Main]: Basepath:  "/home/max/Bureau"
Debug [Main]: "/home/max/Bureau"
Debug [Main]: directory  QUrl("file:///home/max/Bureau")  not in use, marked dirty.
Debug [Main]: aborted, directory  QUrl("file:///home/max/Bureau")  not in cache.
Debug [Main]: aborted, directory  QUrl("file:///home/max")  not in cache.
Debug [Main]: "/home/max/Bureau/default.m3u"
Debug [Main]: aborted, directory  QUrl("file:///home/max/Bureau")  not in cache.
Debug [Main]: aborted, directory  QUrl("file:///home/max/Bureau")  not in cache.
Debug [Main]: "/home/max/Bureau"
Debug [Main]: aborted, directory  QUrl("file:///home/max/Bureau")  not in cache.
Debug [Main]: aborted, directory  QUrl("file:///home/max")  not in cache.
Debug [Main]: ("file:///home/max/Bureau/default.m3u")
Debug [Main]: item not found for QUrl("file:///home/max/Bureau/default.m3u")
Debug [Main]: QUrl("tags:/")
Debug [Main]: QUrl("tags:/")
Debug [Main]: directory  QUrl("tags:/")  not in use, marked dirty.
Debug [Main]: aborted, directory  QUrl("tags:/")  not in cache.
Debug [Main]: QUrl("tags:/")
Debug [Main]: QUrl("tags:/")
Debug [Main]: aborted, directory  QUrl("tags:/")  not in cache.
Debug [Main]: QUrl("tags:/")
Debug [Main]: QUrl("tags:/")
Debug [Main]: aborted, directory  QUrl("tags:/")  not in cache.
```

**Comment by daschuer** [comment]
`Debug [Main]: Already focused on playlist  -1` Indicates that you have selected a not existing playlist (kInvalidPlaylistId). 

The related code has been improved in the 2.4 branch, in PRs like this: 
https://github.com/mixxxdj/mixxx/pull/4193

Please verify if the issue still happens in Mixxx 2.4-beta

**Comment by daschuer** [comment]
Can this be related to https://github.com/mixxxdj/mixxx/issues/11576
Please try to launch Mixxx with: 
`QT_QPA_PLATFORM=wayland mixxx`
Does this fix the issue?

**Comment by jmfergeau** [comment]
> Can this be related to #11576 Please try to launch Mixxx with: `QT_QPA_PLATFORM=wayland mixxx` Does this fix the issue?
Tried and no, it doesn't fix it. I avoid using wayland anyway because nvidia gpu.

> To be more clear: is **crate** export working? Like Playlist items, you can export Crates to popular playlist formats, as well.
Tried to export one of my crates and this didn't worked either. Same problem.

> Please also test playlist **import**. I hooe that helps narrow this down to some specific part of QFileDialog, or Mixxx.
I made a playlist with audacious as m3u and imported it in mixxx. It worked with no issues.

**But, exporting back this very playlist from mixxx worked!**

I'm suspecting my own playlist in mixxx to have a problem. Maybe there's too many files? 600+ is a high number... Maybe it's the foreign characters since i have some japanese tracks that have foreign characters in the file names... but as i said before, this was never a problem for mixxx before and as long as it was a m3u8, it was making a functional playlist.

> What Qt version are you using? It is printed in the log stream, during startup.
```
Debug [Main]: Qt: 5.15.10
Debug [Main]: libshout: 2.4.1
Debug [Main]: PortAudio: 1246976 PortAudio V19.7.0-devel, revision unknown
Debug [Main]: RubberBand: 3.2.1
Debug [Main]: SoundTouch: 2.3.2
Debug [Main]: TagLib: 1.13.0
Debug [Main]: ChromaPrint: 1.5.1
Debug [Main]: Vorbis: Xiph.Org libVorbis 1.3.7
Debug [Main]: libsndfile: libsndfile-1.2.0
Debug [Main]: FLAC: 1.4.3
Debug [Main]: libmp3lame: 3.100
```

**Comment by ronso0** [comment]
Okay, thanks for testing.

What @daschuer spotted in the log (invalid playlist id) indicates something's wrong with the sidebar selection and/or the context menu. (also there are a lot of debug messages, presumably from Qt, that I never saw before in logs, and I'm puzzled by the QFileDialog errors, too).

Let's break this down:
* explicitly select a playlist with left mouse click (its content is shown in the tracks view)
* right-click that item and try to export (attach the log please)
* if that doesn't make any difference and if your desktop has a keyboard shortcut for the context menu, like Shift+F10, also try that

> I don't know for beta. I only use stable versions.

2.4-beta is about to be polished for release within the next few weeks, and it has been tested for quite some time now, so the risk of serious regressions is rather low.
Please give it a shot!

**Comment by jmfergeau** [comment]
> * explicitly select a playlist with left mouse click (its content is shown in the tracks view)
> * right-click that item and try to export (attach the log please)

This is exactly what i'm doing from the beginning. And what i did in my first posts.

Ok this is **really strange** : After some tries I discovered a workaround to make the playlists exportable again as intended. But this workaround needs to be made everytime mixxx is started. Here it is:
- Start Mixxx (trying to export a playlist at this point will fail)
- Import a new playlist. It can be a completely empty playlist. (you can even import the failed exported playlist!)
- Export playlist will now work again with any playlist until you close mixxx.

Here's the log i got from a session where export was successful : [mixxx.log](https://github.com/mixxxdj/mixxx/files/12333478/mixxx.log)

I am very confused now.

**Comment by daschuer** [comment]
Maybe adding an additional tree member fixed the issue?

**Comment by ronso0** [comment]
> Maybe adding an additional tree member fixed the issue?

wtf... that would be a weird bug
@jmfergeau Does it also work if you create a new playlist?
Or if you delete a playlist?
Or if you create a new **crate?**
The former two actions would rebuild the sidebar child model.

Either way, I'm not sure if spending time on a fix is worth it if it works in 2.4
It would also be helpful if someone else on Arch with Qt 5.15.10 could try to reproduce with 2.3 and 2.4. @Holzhaus you're on Arch, too, right? Could you do this?

**Comment by daschuer** [comment]
Yes we have already a release candidate for 2.3.6 which is the last version from the 2.3 branch. So let's focus one 2.4-beta. 
Are you able to build the 2.4 branch yourself? Do you experience the issue with that as well?

**Comment by jmfergeau** [comment]
> wtf... that would be a weird bug @jmfergeau Does it also work if you create a new playlist? Or if you delete a playlist? Or if you create a new **crate?** The former two actions would rebuild the Playlists sidebar child model.

Thanks to your suggestions, I made more tests and I figured out a bit more what might be the cause.

It seems to happen if you start mixxx, stay on the tracks list and attempt to export a playlist from there. And it seems to not occur anymore if you click the playlist to see its content. 

It looks like the playlist exports only if it's first loaded in memory. A thing that doesn't happen when you just started mixxx right away, which is logical.

And same thing for the crates. As long as they aren't "loaded" (displayed on the tracks list) the export will fail. But will succeed if it's loaded.
